spark/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerUpdater.scala

```diff
@@ -17,25 +17,29 @@ class AccumuloLayerUpdater[K: Boundable: JsonFormat: ClassTag, V: ClassTag, Cont
   extends LayerUpdater[LayerId, K, V, Container with RDD[(K, V)]] {
 
   def update(id: LayerId, rdd: Container with RDD[(K, V)]) = {
-    try {
-      if (!attributeStore.layerExists(id)) throw new LayerNotExistsError(id)
-      implicit val sc = rdd.sparkContext
-      implicit val mdFormat = cons.metaDataFormat
+    if (!attributeStore.layerExists(id)) throw new LayerNotFoundError(id)
+    if (rdd.isEmpty()) throw new LayerUpdateError(id, ": empty rdd update")
```

Comment:
```scala
def isEmpty(): Boolean = withScope {
    partitions.length == 0 || take(1).length == 0
  }
```

For rdd it is possible to have one but empty partition. The other option we can set `try` block to wrap `boundable.getKeyBounds` function, and to catch unsupported operation exception: due to `reduce` used in its implementation. To provide a valid function application to the empty rdd we have to make an empty `KeyBounds` instance (to use `foldLeft` instead of `reduce`), @echeipesh also had some ideas about it.